### PR TITLE
feat: implement issues 58-61

### PR DESCRIPTION
**Pull request by ymuichiro**
## Summary
- implement user dictionary editing with CSV import/export in Settings and add Swift coverage
- fix Japanese punctuation post-processing edge cases and expand Python regression tests
- show the active recording input device in the recording indicator and add recording UI coverage
- add voice shortcut storage, normalized matching, key-command execution, and Settings management UI

## Validation
- make test-all
- .venv/bin/ty check python/
- .venv/bin/ruff check python tests/python
- cd KotoType && swift build
- cd KotoType && swift test --skip-build

Closes #58
Closes #59
Closes #60
Closes #61

### CODE REVIEW

File: KotoType/Sources/KotoType/Input/KeystrokeSimulator.swift [hunk 1]
```diff
@@ -142,4 +207,25 @@ final class KeystrokeSimulator {
...
+    private static func orderedModifierSequence(
+        for command: HotkeyConfiguration
+    ) -> [(keyCode: CGKeyCode, flag: CGEventFlags)] {
+        var sequence: [(CGKeyCode, CGEventFlags)] = []
+
+        if command.useControl {
+            sequence.append((0x3B, .maskControl))
+        }
+        if command.useOption {
+            sequence.append((0x3A, .maskAlternate))
```
[the remark this answers is not in the document]
Comment (ymuichiro) [reply KotoType/Sources/KotoType/Input/KeystrokeSimulator.swift]: 対応しました。`HotkeyConfiguration` に記録されている modifier side を再生時にも尊重するように変更し、右側 modifier の key code が使われることをテストでも確認しています。

File: KotoType/Sources/KotoType/Support/UserDictionaryManager.swift [hunk 2]
```diff
@@ -75,17 +101,89 @@ final class UserDictionaryManager: @unchecked Sendable {
...
+                throw UserDictionaryCSVError.invalidFormat(row: index + 1)
+            }
+
+            let rawValue = row[0].trimmingCharacters(in: .whitespacesAndNewlines)
+            if rawValue.isEmpty {
+                blankCount += 1
+                continue
+            }
+
+            if index == 0 && rawValue.caseInsensitiveCompare("term") == .orderedSame {
```
[the remark this answers is not in the document]
Comment (ymuichiro) [reply KotoType/Sources/KotoType/Support/UserDictionaryManager.swift:138]: 対応しました。CSV import 時に先頭セルの UTF-8 BOM を除去してからヘッダー判定するように修正し、BOM 付き `term` ヘッダーをスキップできるテストも追加しました。

File: KotoType/Sources/KotoType/UI/RecordingIndicatorView.swift [hunk 3]
```diff
@@ -45,19 +49,25 @@ struct RecordingIndicatorView: View {
             return CGSize(width: 280, height: 68)
         }
 
+        if state == .recording && !(recordingInputDeviceName?.isEmpty ?? true) {
```
[the remark this answers is not in the document]
Comment (ymuichiro) [reply KotoType/Sources/KotoType/UI/RecordingIndicatorView.swift]: 対応しました。録音中インジケータは入力デバイス名が `nil` の場合でも `Mic:` 行を描画するため、`.recording` のレイアウトを常に高い方に統一してクリップを防ぐようにしました。合わせてレイアウトテストも更新しています。

File: KotoType/Sources/KotoType/UI/SettingsView.swift [hunk 4]
```diff
@@ -54,6 +55,15 @@ struct SettingsView: View {
...
     @State private var pendingDictionaryEntry: String
+    @State private var dictionaryStatusMessage: String?
+    @State private var dictionaryStatusMessageIsError = false
+    @State private var voiceShortcuts: [VoiceShortcut]
+    @State private var pendingVoiceShortcutTrigger: String
+    @State private var pendingVoiceShortcutActionKind: VoiceShortcutActionKind
+    @State private var pendingVoiceShortcutInsertText: String
+    @State private var pendingVoiceShortcutKeyCommand: HotkeyConfiguration
+    @State private var voiceShortcutStatusMessage: String?
+    @State private var voiceShortcutStatusMessageIsError = false
```
[the remark this answers is not in the document]
Comment (ymuichiro) [reply KotoType/Sources/KotoType/UI/SettingsView.swift]: 対応しました。今回の実装では voice shortcut のステータスメッセージにエラー状態を載せていなかったため、未使用だった `voiceShortcutStatusMessageIsError` と条件分岐を削除して状態を整理しました。

File: KotoType/Sources/KotoType/Support/VoiceShortcutManager.swift [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+        normalized = stripEdgeSymbols(from: normalized)
+        normalized = stripTrailingSentencePunctuation(from: normalized)
+        normalized = stripEnclosingPairs(from: normalized)
+        normalized = stripEdgeSymbols(from: normalized)
+        normalized = collapseWhitespace(in: normalized)
+
+        return normalized
+            .folding(
+                options: [.caseInsensitive, .diacriticInsensitive, .widthInsensitive],
+                locale: .current
```
[the remark this answers is not in the document]
Comment (ymuichiro) [reply KotoType/Sources/KotoType/Support/VoiceShortcutManager.swift]: 対応しました。shortcut の正規化はユーザー環境に依存しない方が安全なので、`Locale(identifier: "en_US_POSIX")` を使う形に変更して、マッチングと重複排除を決定的にしました。
